### ISSUE TITLE
calculate zip code characteristics

### ISSUE THREAD

**Issue by wibeasley** [National-COVID-Cohort-Collaborative/CS-Rural-Health#3]
In a discussion today with @jerrodanzalone, we thought we'd calculate some zip/zcta characteristics to make available in the Enclave as a static dataset with 41k rows.  Variables such as
- [x] straight-line distance to the nearest hospital
- [x] ruca?
- [ ] Census demographics

**Comment by jerrodanzalone** [comment]
Given we have access to row-level patient data for demographics, I don't see that being terribly useful. Thanks, Will! I'll put together the accompanying spreadsheet later this week.

**Comment by jerrodanzalone** [comment]
@wibeasley This looks great. As a secondary analysis to compare differences in outcome between access to acute care and critical access, would it be possible to have an additional dataset that indicates distance from ZIP Code to those as separate variables? No rush at all on this as I'm having the current iteration imported into N3C. Thanks!

**Comment by wibeasley** [comment]
@jerrodanzalone, do you have a preference between long/wide?  Here's long (semi-fake) example w/ one row per [zip] x [hospital type].

```
A tibble: 98 x 8
 zip_code hospital_type      distance_min count_within_20 count_within_60 count_within_100 zip_code_prefix~ year_last_exist~
 <chr>    <chr>                     <int>           <int>           <int>            <int> <chr>                       <int>
 00801    critical access               1               1               3               31 008                          2020
 00801    general acute care            4               1               8               45 008                          2020
 01086    critical access              64               0               0                5 010                          2020
 01086    general acute care            2               5              44              166 010                          2020
 01199    critical access              64               0               0                4 011                          2020
 01199    general acute care            0               8              47              160 011                          2020
# ... with 88 more rows
```

Or would you prefer one row per zip (with variables like `distance_min_acute_care` and `distance_min_critical_access`)?  It's 5 lines of R code [to pivot wide](https://tidyr.tidyverse.org/reference/pivot_wider.html), so just tell me your preference.

**Comment by jerrodanzalone** [comment]
Hi @wibeasley. I would end up making one row per zip, so whatever is easier for you to put together. Thanks!

**Comment by jerrodanzalone** [comment]
@wibeasley I think this looks great. Thank you! In terms of methods, each degree of lat/long is approximately 69 miles. Did you set the window to 2 or 3 degrees?

**Comment by wibeasley** [comment]
Currently it's +/-3 latitude and +/-4 longitude.  That's arbitrary, and I'm happy to widen it.

I eyeballed the zips that are missing distances --they are places like Puerto Rico and northern Alaska.  If we expanded the boundaries, I'm not sure distance remains relevant for those zips.  I think a missing value for `distance_min_critical` in Puerto Rico is more representative than a providing the distance to Miami or Anchorage.

https://github.com/National-COVID-Cohort-Collaborative/CS-Rural-Health/blob/c7d5a5ecd62002d0f88d8aa29173488422136684/zip-code-characteristics/manipulation/zip-code-hospital.R#L122-L126

The live version of the file is:  https://github.com/National-COVID-Cohort-Collaborative/CS-Rural-Health/blob/main/zip-code-characteristics/manipulation/zip-code-hospital.R
